lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -335,48 +335,21 @@ class window.Turbolinks
   bypassOnLoadPopstate = (fn) ->
     setTimeout fn, 500
 
-  browserCompatibleDocumentParser = ->
-    createDocumentUsingParser = (html) ->
-      (new DOMParser).parseFromString html, 'text/html'
-
-    createDocumentUsingDOM = (html) ->
-      doc = document.implementation.createHTMLDocument ''
-      doc.documentElement.innerHTML = html
-      doc
-
-    createDocumentUsingWrite = (html) ->
-      doc = document.implementation.createHTMLDocument ''
-      doc.open 'replace'
-      doc.write html
-      doc.close()
-      doc
-
-    # Use createDocumentUsingParser if DOMParser is defined and natively
-    # supports 'text/html' parsing (Firefox 12+, IE 10)
-    #
-    # Use createDocumentUsingDOM if createDocumentUsingParser throws an exception
-    # due to unsupported type 'text/html' (Firefox < 12, Opera)
-    #
-    # Use createDocumentUsingWrite if:
-    #  - DOMParser isn't defined
-    #  - createDocumentUsingParser returns null due to unsupported type 'text/html' (Chrome, Safari)
-    #  - createDocumentUsingDOM doesn't create a valid HTML document (safeguarding against potential edge cases)
-    try
-      if window.DOMParser
-        testDoc = createDocumentUsingParser '<html><body><p>test'
-        createDocumentUsingParser
-    catch e
-      testDoc = createDocumentUsingDOM '<html><body><p>test'
-      createDocumentUsingDOM
-    finally
-      unless testDoc?.body?.childNodes.length is 1
-        return createDocumentUsingWrite
+  createDocument = (html) ->
+    if /<(html|body)/i.test(html)
+      doc = document.documentElement.cloneNode()
+      doc.innerHTML = html
+    else
+      doc = document.documentElement.cloneNode(true)
+      doc.querySelector('body').innerHTML = html
```

Comment:
This is taken straight out of turbolinks master so I think there's merit to leaving it as-is - or adding comments at most. That said I'm not 100% sure either but it looks like the else half does a deep clone of the document's root element because `<body` or `<html` are not present in the argument html, and sets the argument html to be a child of the copied current document's `<body>`, whereas the if half just clones the root node and sets the passed-in html as the child of that.